grin/grin.py

```diff
@@ -412,7 +412,15 @@ def main(argv=None):
         g = GrepText(regex, args)
         openers = dict(text=open, gzip=gzip.open)
         for filename, kind in get_filenames(args):
-            report = g.grep_a_file(filename, opener=openers[kind], encoding=args.encoding)
+            try:
+                report = g.grep_a_file(filename, opener=openers[kind], encoding=args.encoding)
+            except (gzip.BadGzipFile, EOFError) as ex:
```

Comment:
`gzip.BadGzipFile` is only available from python 3.8, so I switch to old `OSError`